x/superfluid/keeper/stake.go

```diff
@@ -503,17 +504,34 @@ func (k Keeper) forceUndelegateAndBurnOsmoTokens(ctx sdk.Context,
 	if err != nil {
 		return err
 	}
-	// TODO: Better understand and decide between ValidateUnbondAmount and SharesFromTokens
-	// briefly looked into it, did not understand what's correct.
-	// TODO: ensure that intermediate account has at least osmoAmount staked.
+
 	shares, err := k.sk.ValidateUnbondAmount(
 		ctx, intermediaryAcc.GetAccAddress(), valAddr, osmoAmount,
 	)
 	if err == stakingtypes.ErrNoDelegation {
 		return nil
-	} else if err != nil {
-		return err
 	}
+
+	if err != nil {
+		// if ValidateUnbondAmount has failed it indicates that the amount we're trying to unbond is
+		// greater then what validator has this can be due to different factors (e.g jail)
+		// in this case we brun min(amount_tpo_burn, total_delegation_share)
```

Comment:
```suggestion
		// in this case we run min(amount_to_burn, total_delegation_share)
```

I think that is what was meant?